test/test-integration/src/test/java/org/wso2am/micro/gw/tests/context/MgwServerInstance.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2am.micro.gw.tests.context;
+
+import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.wso2am.micro.gw.tests.common.BaseTestCase;
+import org.wso2am.micro.gw.tests.mockbackend.MockBackendServer;
+import org.wso2am.micro.gw.tests.util.HttpClientRequest;
+import org.wso2am.micro.gw.tests.util.HttpResponse;
+import org.wso2am.micro.gw.tests.util.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Mgw server instance class.
+ */
+public class MgwServerInstance implements MgwServer {
+
+    private static final Logger log = LoggerFactory.getLogger(MgwServerInstance.class);
+    private DockerComposeContainer environment;
+
+
+    /**
+     * initialize a docker environment using docker compose.
+     *
+     * @throws IOException
+     * @throws MicroGWTestException
+     */
+    public MgwServerInstance() throws IOException, MicroGWTestException {
+        createTmpMgwSetup();
+        File targetClassesDir = new File(MgwServerInstance.class.getProtectionDomain().getCodeSource().
+                getLocation().getPath());
+        String mgwServerPath = targetClassesDir.getParentFile().toString() + File.separator + "server-tmp";
+
+        String dockerCompsePath = mgwServerPath+  File.separator + "docker-compose.yaml";
```

Comment:
Formatting issue: Add a space between **mgwServerPath** and **+**

Reply:
Fixed in 3b6cc6755810d748cd3de4fb25696a5692c08e78